modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteTrimExchange.java

```diff
@@ -87,8 +94,10 @@ public IgniteTrimExchange(RelInput input) {
 
     /** {@inheritDoc} */
     @Override public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
-        double rowCount = mq.getRowCount(this);
-        return planner.getCostFactory().makeCost(rowCount, rowCount, 0);
+        double rowCount = mq.getRowCount(getInput());
+
+        return planner.getCostFactory().makeCost(rowCount,
+            rowCount * (IgniteCost.ROW_COMPARISON_COST + IgniteCost.ROW_PASS_THROUGH_COST), 0);
```

Comment:
not obvious here, can u explain plz ? why : IgniteCost.ROW_COMPARISON_COST + IgniteCost.ROW_PASS_THROUGH_COST

Reply:
TrimExchange just filter out rows which don't met the distribution function's. So the cost function is the same as for IgniteFilter